pandas/core/indexes/datetimelike.py

```diff
@@ -88,9 +80,51 @@ class DatetimeIndexOpsMixin(NDArrayBackedExtensionIndex):
     _is_numeric_dtype = False
     _can_hold_strings = False
     _data: DatetimeArray | TimedeltaArray | PeriodArray
-    freq: BaseOffset | None
-    freqstr: str | None
-    _resolution_obj: Resolution
+
+    # ------------------------------------------------------------------------
+
+    @doc(DatetimeLikeArrayMixin.mean)
+    def mean(self, *, skipna: bool = True, axis: int | None = 0):
+        return self._data.mean(skipna=skipna, axis=axis)
+
+    # error: Decorated property not supported
+    @property  # type: ignore[misc]
+    @doc(DatetimeLikeArrayMixin.asi8)
+    def asi8(self) -> npt.NDArray[np.int64]:
+        return self._data.asi8
+
+    # error: Decorated property not supported
+    @property  # type: ignore[misc]
+    @doc(DatetimeLikeArrayMixin.freq)
+    def freq(self) -> BaseOffset | None:
+        return self._data.freq
+
+    @freq.setter
+    def freq(self, value) -> None:
```

Comment:
do we allow setting?

Reply:
It works at least on main `pd.TimedeltaIndex([]).freq = None` but I don't know whether that is meant to be the case.

Happy to remove the setter (seems like that would be something that is not possible with `inherit_names`: it would copy both the getter and setter),

Reply:
It cannot be set for `PeriodArray` but other arrays (and index) types support it.

pandas/tests/arrays/categorical/test_constructors.py has two tests that set freq to None (but the tests also pass when removing those lines).

Reply:
I moved it to `DatetimeTimedeltaMixin`